Maze.cpp

```diff
@@ -17,17 +18,101 @@ const CellCoordinate Maze::CELL_FINISH =
 void Maze::reset() {
     for (int y = 0; y < NODE_ROWS; y++) {
         for (int x = 0; x < NODE_COLS; x++) {
-            maze[y][x] = Node(NodeCoordinate(x, y));
+            NodeCoordinate pos = NodeCoordinate(x, y);
+            maze[y][x] = Node(pos);
 
             // Initialize border
-            if (x == 0 || y == 0 || y == NODE_ROWS - 1 || x == NODE_COLS - 1)
-                setWall(NodeCoordinate(x, y));
+            if (isBorder(pos) || pos.isPost())
+                setWall(pos);
         }
     }
-
+    setWall(NodeCoordinate(NODE_COLS / 2, NODE_ROWS / 2), false);
     path.clear();
 }
 
+bool Maze::isBorder(NodeCoordinate pos) {
+    return pos.x == 0 || pos.y == 0 || pos.y == NODE_ROWS - 1 ||
+           pos.x == NODE_COLS - 1;
+}
+
+bool Maze::isBorder(Node node) {
+    return isBorder(node.pos);
+}
+
+bool Maze::withinBounds(NodeCoordinate pos) {
+    return pos.x >= 0 && pos.x < NODE_COLS && pos.y >= 0 && pos.y < NODE_ROWS;
+}
+
+bool Maze::withinBounds(Node node) {
+    return withinBounds(node.pos);
+}
+
+void Maze::generate(int seed) {
+    reset();
+    srand(seed);
+    vector<CellCoordinate> backtrackStack;
+    set<CellCoordinate> unvisited;
+    backtrackStack.reserve(CELL_COLS * CELL_ROWS);
+
+    for (int i = 0; i < CELL_ROWS; i++) {
+        for (int j = 0; j < CELL_COLS; j++) {
```

Comment:
Minor thing but if you're iterating over x and y, why not name it `x` and `y` to be more clear.